homeassistant/components/eight_sleep/__init__.py

```diff
@@ -170,9 +199,25 @@ def __init__(
 
         mapped_name = NAME_MAP.get(sensor, sensor.replace("_", " ").title())
         if self._user_obj is not None:
-            mapped_name = f"{self._user_obj.side.title()} {mapped_name}"
-
-        self._attr_name = f"Eight {mapped_name}"
-        self._attr_unique_id = (
-            f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
-        )
+            assert self._user_obj.user_profile
+            name = f"{self._user_obj.user_profile['firstName']}'s {mapped_name}"
+            self._attr_name = name
+        else:
+            self._attr_name = f"Eight Sleep {mapped_name}"
+        unique_id = f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
+        self._attr_unique_id = unique_id
+        identifiers = {(DOMAIN, _get_device_unique_id(eight, self._user_obj))}
+        self._attr_device_info = DeviceInfo(identifiers=identifiers)
+
+    async def async_heat_set(self, target: int, duration: int) -> None:
```

Comment:
Why do we put this method and service on the entity? It doesn't seem to require an entity. I'd expect it to target a user representation.

Reply:
See my last comment - this was how the service was already set up, I just made it a proper entity service because the existing logic wouldn't have worked.